graylog2-web-interface/src/views/logic/views/UseCreateViewForEvent.ts

```diff
@@ -0,0 +1,200 @@
+    res.aggregationWidgets.push(widget);
+    res.aggregationTitles[widgetId] = title;
+    res.aggregationPositions[widgetId] = position;
+
+    if (needsSummaryAggregations) {
+      summaryAggregations.fnSeries.push(fnSeries);
+      summaryAggregations.rowPivots.push(field);
+      summaryAggregations.title = `${summaryAggregations.title} ${title}`;
+    }
+
+    return res;
+  }, { aggregationTitles: {}, aggregationWidgets: [], aggregationPositions: {} });
+
+  const widgets = [
+    ...aggregationWidgets,
+    histogram,
+    messageTable,
+  ];
+
+  const titles = {
+    widget: {
+      ...aggregationTitles,
+      [histogram.id]: 'Message Count',
+      [messageTable.id]: 'All Messages',
+    },
+  };
+
+  const positions = {
+    ...aggregationPositions,
+    [histogram.id]: new WidgetPosition(1, AGGREGATION_WIDGET_HEIGHT * aggregationWidgets.length + 1 + SUMMARY_ROW_DELTA, 2, Infinity),
+    [messageTable.id]: new WidgetPosition(1, AGGREGATION_WIDGET_HEIGHT * aggregationWidgets.length + 3 + SUMMARY_ROW_DELTA, 6, Infinity),
+  };
+
+  if (needsSummaryAggregations) {
+    const summaryAggregationWidget = getAggregationWidget({
+      rowPivots: [pivotForField(uniq([...summaryAggregations.rowPivots, ...groupBy]), new FieldType('value', [], []))],
+      fnSeries: summaryAggregations.fnSeries.map((s) => Series.forFunction(s)),
+    });
+    widgets.push(summaryAggregationWidget);
+    titles.widget[summaryAggregationWidget.id] = summaryAggregations.title;
+    positions[summaryAggregationWidget.id] = new WidgetPosition(1, 1, AGGREGATION_WIDGET_HEIGHT, Infinity);
+  }
+
+  return { titles, widgets, positions };
+};
+
+export const ViewStateGenerator = async ({ streams, aggregations, groupBy }: {groupBy: Array<string>, streams: string | string[] | undefined, aggregations: Array<any>}) => {
+  const { titles, widgets, positions } = await WidgetsGenerator({ streams, aggregations, groupBy });
+
+  return ViewState.create()
+    .toBuilder()
+    .titles(titles)
+    .widgets(Immutable.List(widgets))
+    .widgetPositions(positions)
+    .build();
+};
+
+export const ViewGenerator = async ({
+  streams,
+  timeRange,
+  queryString,
+  aggregations,
+  groupBy,
+}: {
+  streams: string | string[] | undefined | null,
+  timeRange: AbsoluteTimeRange | RelativeTimeRangeStartOnly,
+  queryString: ElasticsearchQueryString,
+  aggregations: Array<EventDefinitionAggregation>
+  groupBy: Array<string>
+},
+) => {
+  const query = QueryGenerator(streams, undefined, timeRange, queryString);
+  const search = Search.create().toBuilder().queries([query]).build();
+  const viewState = await ViewStateGenerator({ streams, aggregations, groupBy });
+
+  const view = View.create()
+    .toBuilder()
+    .newId()
+    .type(View.Type.Search)
+    .state({ [query.id]: viewState })
+    .search(search)
+    .build();
+
+  return UpdateSearchForWidgets(view);
+};
+
+export const UseCreateViewForEvent = (
```

Comment:
usually the cutom hooks start lowercase.

Reply:
We have a pretty similar hook here `src/views/logic/views/UseCreateSavedSearch.ts` and it starts with uppercase. So I decided to make them similar. If you think that would be better to have lowercase here I can change it also in `UseCreateSavedSearch`.

Reply:
I see!